wagon-providers/wagon-http-shared/src/main/java/org/apache/maven/wagon/shared/http/AbstractHttpClientWagon.java

```diff
@@ -354,6 +360,92 @@ private static PoolingHttpClientConnectionManager createConnManager()
         return connManager;
     }
 
+    /**
+     * The type of the retry handler, default to DefaultHttpRequestRetryHandler.
+     * Values can be default, standard (StandardHttpRequestRetryHandler), or a fully qualified name class.
+     *
+     * @since 3.2
+     */
+    private static final String RETRY_HANDLER_CLASS =
+            System.getProperty( "maven.wagon.http.retryhandler.class", "standard" );
+
+    /**
+     * true if it's OK to retry non-idempotent requests that have been sent.
```

Comment:
This statement is not true. The original documentation does not say this. Even the default implementation will stop before this value is read if the request ist not idempotent.